events.js

```diff
@@ -0,0 +1,62 @@
+/* eslint no-plusplus: ["error", { "allowForLoopAfterthoughts": true }]
+*/
+import { sortList, display, updateToDoList } from './toDoList.js';
+import status from './status.js';
+import crud from './crud.js';
+
+const events = (() => {
+  const updateList = (toDoList, parent, dragTarget, target, upwards) => {
+    if (upwards) {
+      parent.insertBefore(dragTarget.parentNode, target.parentNode);
+      toDoList[dragTarget.dataset.id].index = toDoList[target.dataset.id].index;
+      for (let i = target.dataset.id; i < dragTarget.dataset.id; ++i) {
+        toDoList[i].index += 1;
+      }
+    } else {
+      parent.insertBefore(target.parentNode, dragTarget.parentNode);
+      toDoList[dragTarget.dataset.id].index = toDoList[target.dataset.id].index;
+      for (let i = target.dataset.id; i > dragTarget.dataset.id; --i) {
+        toDoList[i].index -= 1;
+      }
+    }
+  };
+  const setDragStart = (toDoList) => {
+    let dragTarget;
+    let initialY;
+    document.addEventListener('drag', () => {
+
+    }, false);
+
+    document.addEventListener('dragstart', (event) => {
+      event.stopPropagation();
+      dragTarget = event.target;
+      initialY = event.clientY;
+    }, true);
+
+    document.addEventListener('dragover', (event) => {
+      event.preventDefault();
+    }, false);
+
+    document.addEventListener('drop', (event) => {
+      event.stopImmediatePropagation();
+      if (event.target.classList.contains('can-swap')) {
+        const parent = document.querySelector('.to-do-list-wrap');
```

Comment:
### Keep DOM access to a minimum    :red_circle:
---
- _Accessing the DOM in browsers is an expensive thing to do. The DOM is a very complex API and rendering in browsers can take up a lot of time. You can see this when running complex web applications when your computer is already maxed out with other work — changes take longer or get shown halfway through and so on._-
---
### Fix

** To make sure that your code is fast and doesn’t slow down the browser to a halt trying to keep DOM access to a bare minimum. Instead of constantly creating and applying elements, have a tool function that turns a string into DOM elements and calls this function at the end of your generation process to disturb the browser rendering once rather than continually.**

Reply:
You are totally right. I forgot about how expensive traversing the DOM is. Changed!